packages/expo-error-recovery/src/ErrorRecovery.ts

```diff
@@ -0,0 +1,39 @@
+import ExpoErrorRecovery from './ExpoErrorRecovery';
+import { Platform } from 'react-native';
+
+let recoveredPropsToSave: String | null = null;
+let globalHandler;
+if (Platform.OS !== 'web') {
+  globalHandler = ErrorUtils.getGlobalHandler();
+  // ErrorUtils came from react-native
+  // https://github.com/facebook/react-native/blob/1151c096dab17e5d9a6ac05b61aacecd4305f3db/Libraries/vendor/core/ErrorUtils.js#L25
+  ErrorUtils.setGlobalHandler(async (error, isFatal) => {
```

Comment:
Hm… This sounds like a side effect. Shouldn't we put it in a separate `.fx.ts` file, like eg. [here](https://github.com/expo/expo/blob/master/packages/expo-asset/src/Asset.fx.ts) and add it to `sideEffects` like [here](https://github.com/expo/expo/pull/4706/files#diff-167ac1570d608a4f8e011cc3bc5c6defR7)?

see https://github.com/expo/expo/pull/4706